src/main/java/org/opensearch/flowframework/workflow/CreateIndex/CreateIndexStep.java

```diff
@@ -0,0 +1,115 @@
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.workflow.CreateIndex;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.Resources;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.admin.indices.create.CreateIndexRequest;
+import org.opensearch.action.admin.indices.create.CreateIndexResponse;
+import org.opensearch.client.Client;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.flowframework.workflow.WorkflowData;
+import org.opensearch.flowframework.workflow.WorkflowStep;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Step to create an index
+ */
+public class CreateIndexStep implements WorkflowStep {
+
+    private static final Logger logger = LogManager.getLogger(CreateIndexStep.class);
+    private Client client;
+    private final String NAME = "create_index_step";
+
+    /**
+     * Instantiate this class
+     * @param client Client to create an index
+     */
+    public CreateIndexStep(Client client) {
+        this.client = client;
+    }
+
+    @Override
+    public CompletableFuture<WorkflowData> execute(List<WorkflowData> data) {
+        CompletableFuture<WorkflowData> future = new CompletableFuture<>();
+        ActionListener<CreateIndexResponse> actionListener = new ActionListener<>() {
+
+            @Override
+            public void onResponse(CreateIndexResponse createIndexResponse) {
+                logger.info("created index:{}", createIndexResponse.index());
+                future.complete(new WorkflowData() {
+                    @Override
+                    public Map<String, Object> getContent() {
+                        return Map.of("index-name", createIndexResponse.index());
+                    }
+                });
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.error("Failed to create an index", e);
+                future.completeExceptionally(e);
+            }
+        };
+
+        String index = null;
+        String type = null;
+        Settings settings = null;
+
+        for (WorkflowData workflowData : data) {
+            Map<String, Object> content = workflowData.getContent();
+            index = (String) content.get("index-name");
+            type = (String) content.get("type");
+            if (index != null && type != null && settings != null) {
+                break;
+            }
+        }
+
+        // TODO:
+        // 1. Create settings based on the index settings received from content
+
+        try {
+            CreateIndexRequest request = new CreateIndexRequest(index).mapping(
+                getIndexMappings("mappings/" + type + ".json"),
```

Comment:
Not blocking for this PR, but in my opinion index mappings should be provided by the user, since the user may want control over what fields they need. Having a static mapping is fine for now as we work towards building support for default use cases, but we'll definitely need some common mechanism to create these request bodies given the provided workflow data. This would apply to pipeline configurations and index settings as well, pretty much for any content that requires us to map fields with the user inputs.

Reply:
I think we can have a mix here... static mappings for well-known things, user-defined mappings otherwise.  We'd just need to have some sort of syntax for the template, e.g., if user puts `{{knn}}` it will use that static mapping, or they can provide their own.